app/models/submission_rule.rb

```diff
@@ -11,6 +11,9 @@ def initialize(rule_name)
   accepts_nested_attributes_for :periods, allow_destroy: true
   validates_associated :periods
   validates :assignment, uniqueness: true
+  validates :penalty_type,
```

Comment:
This is okay, but please override this validation in the two penalty-based submission rules to disallow `nil`. For those two rules we should always expect a non-nil penalty type.